### ISSUE TITLE
Add heat map to float32, float16, int16, and int8 dtypes

### ISSUE THREAD

**Issue by Chuck321123** [spyder-ide/spyder#21682]
Didnt realize there were no heat map/ background colors for the dtypes mentioned in the title. See for yourself.

```python

import pandas as pd
import numpy as np

# Set a random seed for reproducibility
np.random.seed(42)

# Create a DataFrame with 50 rows and two columns
data = {
    'Col1': np.random.randint(1, 1000, 50),
    'Col2': np.random.uniform(1, 1000, 50).round(3)         
}

x = pd.DataFrame(data)

x["Col3"] = x["Col2"].astype("float64")
x["Col4"] = x["Col2"].astype("float32")
x["Col5"] = x["Col2"].astype("float16")
x["Col6"] = x["Col1"].astype("int64")
x["Col7"] = x["Col1"].astype("int32")
x["Col8"] = x["Col1"].astype("int16")
x["Col9"] = x["Col1"].astype("int8")
```

**Comment by jitseniesen** [comment]
Hi, @Chuck321123, thanks for the report. This is fixed in the upcoming Spyder 6 release which I think should become available in two months or so. The fix is in PR #21295. It's actually a lucky side-effect of that change, as I did not have the smaller float and int types in mind when writing it. The only potential issue is that the fix requires Pandas version 2. Would that be a problem for you?

**Comment by Chuck321123** [comment]
@jitseniesen No, thats perfect. Thanks
